pkg/auth/tanzu/kubeconfig.go

```diff
@@ -18,6 +19,14 @@ import (
 	configtypes "github.com/vmware-tanzu/tanzu-plugin-runtime/config/types"
 )
 
+const (
+	// TanzuLocalKubeDir is the local config directory
+	TanzuLocalKubeDir = ".kube-tanzu"
+
+	// TanzuKubeconfigFile is the name the of the kubeconfig file
+	TanzuKubeconfigFile = "config"
+)
```

Comment:
@vuil I had choosen to user `$HOME/.kube-tanzu/config` as the internal kubeconfig file to store kubeconfig of TAP SaaS contexts

Reply:
There was an ask to expose all the configuration locations `tanzu` cli uses and using different locations for different configurations might be difficult for the user to understand if the user needs to know.

So, Should we consider using the `~/.config/tanzu` location to store all our configurations including this kube config? I don't strongly feel about one or another but It might be better to keep everything in one place from the user's perspective.

Reply:
I don't feel too strongly myself, but I lean toward's @anujc25's suggestion for the same reason. The only disadvantage I can think of is what happens when a user want to start afresh and on a new set of tanzu config (and thinks that wiping out everything in ~/.config/tanzu is the way to go) may lose those kube contexts as well. That feels like a very corner case, and not enough to sway my opinion.

Reply:
How about something like `~/.config/tanzu/kube/config.yaml` if we go with location within `~/.config/tanzu`?

Reply:
> he only disadvantage I can think of is what happens when a user want to start afresh on a new set of tanzu config (and thinks that wiping out everything in ~/.config/tanzu is the way to go). In that situation, use could lose those kube contexts as well. That feels like a very corner case, and not enough to sway my opinion.

Yes, I am not worried about it much because the generated kubeconfig has exec plugin which refers to context which is part of ~/.config directory anyways.

So, I am fine using `~/.config/tanzu/kube/config` (without .yaml extension)

Reply:
Updated the PR to use  `~/.config/tanzu/kube/config kubectl` as internal kubeconfig file